### PR TITLE
Update 1.13

### PR DESCRIPTION
**Pull request by ash2k**
This PR is a 
 - [x] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

**What this PR does / why we need it**:
Update kubernetes libraries to 1.13.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,6 +154,9 @@ generators: $(GENERATORS)
 
 .SECONDEXPANSION:
 
+$(BINDIR)/openapi-gen: $$(shell find vendor/k8s.io/kube-openapi/cmd/openapi-gen vendor/k8s.io/gengo) .init
```
Comment (ash2k) [inline Makefile]: `openapi-gen` now lives in its own repo

File: cmd/apiserver/app/server/util.go [hunk 2]
```diff
@@ -91,7 +91,13 @@ func buildGenericConfig(s *ServiceCatalogServerOptions) (*genericapiserver.Recom
 		genericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysAllowAuthorizer()
 	}
 
-	if err := s.AuditOptions.ApplyTo(&genericConfig.Config); err != nil {
+	if err := s.AuditOptions.ApplyTo(
+		&genericConfig.Config,
+		genericConfig.ClientConfig,
+		genericConfig.SharedInformerFactory,
+		// TODO where to get namespace? Downward api?
```
Comment (ash2k) [inline cmd/apiserver/app/server/util.go]: Not sure what to do here

Comment (ash2k) [reply cmd/apiserver/app/server/util.go]: https://github.com/kubernetes-sigs/controller-runtime/blob/50b6bdd0660bf82fdedc64bae06e5a2e878685bc/pkg/leaderelection/leader_election.go#L93

**Comment by ash2k** [comment]
@sttts hey, could you help me understand this openapi generator failure? I have spent an hour trying to understand it, but failed. It works fine on my mac laptop (service catalog uses docker for local builds though) but travis consistently fails. See https://travis-ci.org/kubernetes-incubator/service-catalog/jobs/468273448#L655-L661

I was not expecting any openapi-related issues because I haven't touched the types in this PR. The only reason can be openapi generator or maybe it is used incorrectly? We invoke it [here](https://github.com/kubernetes-incubator/service-catalog/blob/13a5b03321c7df3a4deaa2fb7ce316eb622e3110/build/update-apiserver-gen.sh#L84).

p.s. also the warnings above look suspicious to me, although I believe they are unrelated. A bug in `ObjectMeta`?
```
I1215 02:11:26.038958      44 openapi.go:395] [k8s.io/apimachinery/pkg/apis/meta/v1.LabelSelectorRequirement] Key string: tag patchMergeKey on type Builtin; only allowed on type Slice
I1215 02:11:26.038986      44 openapi.go:395] [k8s.io/apimachinery/pkg/apis/meta/v1.LabelSelectorRequirement] Key string: tag patchStrategy on type Builtin; only allowed on type Slice
```

**Comment by ash2k** [comment]
Hm, so `make .generate_files` works fine locally, but CI runs `make verify` which fails. I was trying to reproduce with the later. The former command fails locally for me so I can test ideas faster. `make verify-generated` is to run generated code verification directly.

**Comment by MHBauer** [comment]
/cc @duglin 
/assign @jberkhahn 
I think Jonathan should look at this to learn how the build works.

**Comment by nilebox** [comment]
> I0102 22:00:17.124] Starting the API Server
I0102 22:00:17.278] 77447588207e7d3460b5e87cd961f481582c4d8632aff960c74430ec10b5c4a6
I0102 22:00:17.717] Waiting for API Server to be available...
I0102 22:00:48.303] Timed-out waiting for API Server

https://storage.googleapis.com/kubernetes-jenkins/logs/pull/kubernetes-incubator_service-catalog/2528/pull-service-catalog-integration/610/build-log.txt

Looks like Service Catalog API server fails to start for some reason...

**Comment by nilebox** [comment]
Something is wrong with the logging flags...
@ash2k I suppose it may be due to `glog` and `klog` conflicting with each other?
```
bin/service-catalog flag redefined: log_dir
panic: bin/service-catalog flag redefined: log_dir

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc000070180, 0x20c6680, 0x31d9af0, 0x198b339, 0x7, 0x19ca233, 0x2f)
	/usr/local/go/src/flag/flag.go:805 +0x529
flag.(*FlagSet).StringVar(0xc000070180, 0x31d9af0, 0x198b339, 0x7, 0x0, 0x0, 0x19ca233, 0x2f)
	/usr/local/go/src/flag/flag.go:708 +0x8a
github.com/kubernetes-incubator/service-catalog/vendor/k8s.io/klog.InitFlags(0xc000070180)
	/go/src/github.com/kubernetes-incubator/service-catalog/vendor/k8s.io/klog/klog.go:411 +0x7b
main.main()
	/go/src/github.com/kubernetes-incubator/service-catalog/cmd/service-catalog/main.go:31 +0x3f
```

**Comment by nilebox** [comment]
Actually, it seems that `k8s.io/apiserver` invokes the `klog.InitFlags()` function (https://github.com/kubernetes/apiserver/blob/master/pkg/util/logs/logs.go#L36), and we don't need to explicitly call it from the main function...

**Comment by nilebox** [comment]
It seems to me that we should fix the issue with `klog.InitFlags()` upstream by removing the call from the apisever util code and replacing it with an explicit invokation in the `main()`...

**Pull request merged**